### DISCUSSION TITLE
How to use the random_alloy function in dpgen?

### DISCUSSION THREAD

**Discussion by XTUer008** [deepmodeling/dpgen#497] [Q&A] [answered]
I found the description "create_random_alloys" in the dpgen/data/tools/create_random_disturb.py. And some of the presentations about dpgen simply mentioned the "Random Alloy, surface" and "Random Alloy, bulk". How can I set the relevant parameters to use this funcion.
Thanks.

**Comment by AnguseZhang** [comment, marked as answer]
You can use `dpgen init_bulk` to prepare perturbed bulk systems  and `dpgen init_surf` to prepare surf structures. 
Please refer to the [manual](https://github.com/deepmodeling/dpgen) and search for `init_bulk` and `init_surf`

**Reply by XTUer008** [reply]
Thanks for your help. 
It looks like that the perturbation of atoms is based on the origin atomic positions, rather than interchange of positions of different kinds of atoms.  Actually, I want to achieve the latter one.

**Reply by AnguseZhang** [reply]
You can tell your needs more specifically. 
See https://github.com/deepmodeling/dpdata and search "perturb" and "replace". Can the function "replace" meet your needs?

**Reply by XTUer008** [reply]
I try to run a training about amorphous alloys, therefore, I want to creat random alloy structures as inital bulk samples.
I think the "replace" fuction able to meet my requirements, and I will try right away.
Thanks again.

**Comment by jaejae9804** [comment]
Hi, I realize this discussion had been uploaded quite a while ago. Im facing the same problem here. What I need is initial bulk structures with lattice points occupied by 3 types of atoms, randomly. May I ask how you were able to overcome this kind of difficulty?
